### PR TITLE
refactor: use same k8s client in apiserver and controller

### PR DESCRIPTION
**Pull request by Abirdcfly**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes

Fixes #

#### Special notes for your reviewer

Also upgraded github.com/99designs/gqlgen from 1.17.40 to 1.17.44 as the old version is not compatible with the latest go 1.22.

### CODE REVIEW

File: api/base/v1alpha1/worker.go [hunk 1]
```diff
@@ -62,6 +63,7 @@ func (worker Worker) Model() TypedObjectReference {
 		modelNs = *worker.Spec.Model.Namespace
 	}
 	return TypedObjectReference{
+		APIGroup:  pointer.String(GroupVersion.String()),
```
Comment (Abirdcfly) [inline api/base/v1alpha1/worker.go:66]: need double check

File: apiserver/pkg/auth/auth.go [hunk 2]
```diff
@@ -68,35 +65,28 @@ func isBearerToken(token string) (string, bool) {
...
 		return false, "", err
 	}
 
-	ok, found, err := unstructured.NestedBool(u1.Object, "status", "allowed")
-	if err != nil || !found {
-		klog.Warning("not found allowed filed or some errors occurred.")
-		return false, "", err
-	}
-	return ok, u.Name, nil
+	return av.Status.Allowed, u.Name, nil
```
Comment (Abirdcfly) [inline apiserver/pkg/auth/auth.go:89]: need double check

File: apiserver/pkg/client/client.go [hunk 3]
```diff
@@ -47,7 +58,27 @@ func GetClient(idtoken *string) (dynamic.Interface, error) {
...
+		return nil, err
+	}
+	return cli, nil
+}
+
+var (
+	Scheme = runtime.NewScheme()
+)
 
-	cfg.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(50, 60)
```
Comment (Abirdcfly) [inline apiserver/pkg/client/client.go:51]: need add this config?

File: apiserver/pkg/client/client.go [hunk 4]
```diff
@@ -47,7 +58,27 @@ func GetClient(idtoken *string) (dynamic.Interface, error) {
...
-	return dynamic.NewForConfig(cfg)
+func init() {
```
Comment (Abirdcfly) [inline apiserver/pkg/client/client.go:74]: need some comment?

File: apiserver/pkg/common/common_list_resource.go [hunk 5]
```diff
@@ -18,29 +18,29 @@ package common
...
 // ListReources filtering resources based on conditions will modify the original array,
 // so if you want to preserve the original data, make a backup before calling the function
-func ListReources(list *unstructured.UnstructuredList, page, pageSize int, converter ResourceConverter, options ...ResourceFilter) (*generated.PaginatedResult, error) {
+func ListReources(items []client.Object, page, pageSize int, converter ResourceConverter, options ...ResourceFilter) (*generated.PaginatedResult, error) {
 	index, optIndex := 0, 0
-	for i := range list.Items {
+	for i := range items {
 		for optIndex = 0; optIndex < len(options); optIndex++ {
-			if !options[optIndex](&list.Items[i]) {
+			if !options[optIndex](items[i]) {
```
Comment (Abirdcfly) [inline apiserver/pkg/common/common_list_resource.go:32]: double check

File: apiserver/pkg/common/common_list_resource.go [hunk 6]
```diff
@@ -52,12 +52,12 @@ func ListReources(list *unstructured.UnstructuredList, page, pageSize int, conve
...
 
-	h := PageNodeSorter(list.Items)
+	h := PageNodeSorter(items)
 	heap.Init(&h)
 	for cur := 0; h.Len() > 0 && cur < end; cur++ {
-		top := heap.Pop(&h).(unstructured.Unstructured)
+		top := heap.Pop(&h).(client.Object)
 		if cur >= start {
-			node, err := converter(&top)
+			node, err := converter(top)
```
Comment (Abirdcfly) [inline apiserver/pkg/common/common_list_resource.go:60]: double check

File: pkg/evaluation/evaluation.go [hunk 7]
```diff
@@ -71,7 +71,7 @@ func NewRagasDatasetGenerator(ctx context.Context, cli dynamic.Interface, app *v
 		}
 	}
 
-	runapp, err := appruntime.NewAppOrGetFromCache(ctx, cli, app)
+	runapp, err := appruntime.NewAppOrGetFromCache(ctx, nil, app) // FIXME
```
Comment (Abirdcfly) [inline pkg/evaluation/evaluation.go:74]: FIXME

File: pkg/evaluation/evaluation.go [hunk 8]
```diff
@@ -150,7 +150,7 @@ func (eval *RagasDatasetGenerator) Generate(ctx context.Context, csvData io.Read
 		}
 
 		// chat with application
-		out, err := eval.app.Run(ctx, eval.cli, nil, appruntime.Input{Question: ragasRow.Question, NeedStream: false, History: nil})
+		out, err := eval.app.Run(ctx, nil, nil, appruntime.Input{Question: ragasRow.Question, NeedStream: false, History: nil}) // FIXME
```
Comment (Abirdcfly) [inline pkg/evaluation/evaluation.go:153]: FiXME

File: apiserver/pkg/application/application.go [hunk 9]
```diff
@@ -134,15 +132,15 @@ func cr2app(prompt *apiprompt.Prompt, chainConfig *apichain.CommonChainConfig, r
 	return gApp, nil
 }
 
-func app2metadataConverter(objApp *unstructured.Unstructured) (generated.PageNode, error) {
-	return app2metadata(objApp)
+func app2metadataConverter(objApp client.Object) (generated.PageNode, error) {
+	app, ok := objApp.(*v1alpha1.Application)
```
Comment (Abirdcfly) [inline apiserver/pkg/application/application.go:136]: double check

File: apiserver/pkg/application/application.go [hunk 10]
```diff
@@ -219,58 +204,71 @@ func UpdateApplication(ctx context.Context, c dynamic.Interface, input generated
...
+			},
+		},
+		&apichain.RetrievalQAChain{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      *input.Name,
+				Namespace: input.Namespace,
+			},
+		},
+		&apiretriever.KnowledgeBaseRetriever{
+
```
Comment (Abirdcfly) [inline apiserver/pkg/application/application.go:235]: remove this line

File: apiserver/pkg/common/common.go [hunk 11]
```diff
@@ -163,26 +118,20 @@ func GetObjStatus(obj client.Object) string {
 	var (
 		condition v1alpha1.Condition
 	)
-	switch obj.GetObjectKind().GroupVersionKind().Kind {
-	case "Datasource":
-		v := obj.(*v1alpha1.Datasource)
+	switch v := obj.(type) {
```
Comment (Abirdcfly) [inline apiserver/pkg/common/common.go:121]: double check

File: apiserver/pkg/common/common.go [hunk 12]
```diff
@@ -288,3 +236,61 @@ func GetAppCategory(app *v1alpha1.Application) []*string {
...
+		client.InNamespace(input.Namespace),
+	}
+	if input.LabelSelector != nil {
+		l, err := labels.Parse(*input.LabelSelector)
+		if err != nil {
+			return nil, err
+		}
+		opts = append(opts, client.MatchingLabelsSelector{Selector: l})
+	}
+	if *input.FieldSelector != "" {
```
Comment (Abirdcfly) [inline apiserver/pkg/common/common.go:288]: not nil

File: apiserver/pkg/common/common_filter.go [hunk 13]
```diff
@@ -19,42 +19,39 @@ import (
...
-	ResourceFilter    func(*unstructured.Unstructured) bool
-	ResourceConverter func(*unstructured.Unstructured) (generated.PageNode, error)
+	ResourceFilter    func(client.Object) bool
+	ResourceConverter func(client.Object) (generated.PageNode, error)
 )
 
-func FilterByName(name string) ResourceFilter {
-	return func(u *unstructured.Unstructured) bool {
-		return u.GetName() == name
+func FilterByContainerName(name string) ResourceFilter {
```
Comment (Abirdcfly) [inline apiserver/pkg/common/common_filter.go:34]:
```suggestion
func FilterByContainsName(name string) ResourceFilter {
```

File: apiserver/pkg/dataset/dataset.go [hunk 14]
```diff
@@ -92,30 +94,39 @@ func CreateDataset(ctx context.Context, c dynamic.Interface, input *generated.Cr
...
-	if err != nil {
-		return nil, err
-	}
-	return dataset2model(obj)
+	return dataset2model(dataset)
 }
 
-func ListDatasets(ctx context.Context, c dynamic.Interface, input *generated.ListDatasetInput) (*generated.PaginatedResult, error) {
-	listOptions := v1.ListOptions{}
+func ListDatasets(ctx context.Context, c client.Client, input *generated.ListDatasetInput) (*generated.PaginatedResult, error) {
```
Comment (Abirdcfly) [inline apiserver/pkg/dataset/dataset.go:104]: todo: change to ListCommonInput

File: apiserver/pkg/dataset/dataset.go [hunk 15]
```diff
@@ -133,55 +144,77 @@ func ListDatasets(ctx context.Context, c dynamic.Interface, input *generated.Lis
...
 	}
-	return dataset2model(obj)
+	return dataset2model(dataset)
 }
 
-func DeleteDatasets(ctx context.Context, c dynamic.Interface, input *generated.DeleteCommonInput) (*string, error) {
+func DeleteDatasets(ctx context.Context, c client.Client, input *generated.DeleteCommonInput) (*string, error) {
 	none := ""
-	listOptions := v1.ListOptions{}
+	listOptions := &client.DeleteAllOfOptions{
```
Comment (Abirdcfly) [inline apiserver/pkg/dataset/dataset.go:179]: change to commnon method

File: apiserver/pkg/datasource/datasource.go [hunk 16]
```diff
@@ -104,42 +99,20 @@ func datasource2model(obj *unstructured.Unstructured) (*generated.Datasource, er
...
+	datasource.Spec.Description = pointer.StringDeref(input.Description, datasource.Spec.Description)
 
 	// make endpoint
-	endpoint, err := common.MakeEndpoint(ctx, c, generated.TypedObjectReferenceInput{
-		APIGroup:  &datasource.APIVersion,
-		Kind:      datasource.Kind,
-		Name:      datasource.Name,
-		Namespace: &datasource.Namespace,
-	}, input.Endpointinput)
+	endpoint, err := common.MakeEndpoint(ctx, c, datasource, input.Endpointinput)
```
Comment (Abirdcfly) [inline apiserver/pkg/datasource/datasource.go:115]: double check

File: apiserver/pkg/datasource/datasource.go [hunk 17]
```diff
@@ -160,65 +133,37 @@ func CreateDatasource(ctx context.Context, c dynamic.Interface, input generated.
...
 	// update auth secret with owner reference
 	if input.Endpointinput.Auth != nil {
 		// user obj as the owner
-		err := common.MakeAuthSecret(ctx, c, generated.TypedObjectReferenceInput{
-			APIGroup:  &common.CoreV1APIGroup,
-			Kind:      "Secret",
-			Name:      common.MakeAuthSecretName(datasource.Name, "datasource"),
-			Namespace: &input.Namespace,
-		}, input.Endpointinput.Auth, obj)
+		err := common.MakeAuthSecret(ctx, c, input.Namespace, common.MakeAuthSecretName(datasource.Name, "datasource"), input.Endpointinput.Auth, datasource)
```
Comment (Abirdcfly) [inline apiserver/pkg/datasource/datasource.go:144]: doubele check

File: apiserver/pkg/datasource/datasource.go [hunk 18]
```diff
@@ -160,65 +133,37 @@ func CreateDatasource(ctx context.Context, c dynamic.Interface, input generated.
...
 
 	// Update endpoint
 	if input.Endpointinput != nil {
-		endpoint, err := common.MakeEndpoint(ctx, c, generated.TypedObjectReferenceInput{
-			APIGroup:  &datasource.APIVersion,
-			Kind:      datasource.Kind,
-			Name:      datasource.Name,
-			Namespace: &datasource.Namespace,
-		}, *input.Endpointinput)
+		endpoint, err := common.MakeEndpoint(ctx, c, datasource, *input.Endpointinput)
```
Comment (Abirdcfly) [inline apiserver/pkg/datasource/datasource.go:166]: double check

File: apiserver/pkg/gpt/gpt.go [hunk 19]
```diff
@@ -103,15 +92,14 @@ func GetGPT(ctx context.Context, c dynamic.Interface, name string) (*generated.G
...
 	keyword := pointer.StringDeref(input.Keyword, "")
 	category := pointer.StringDeref(input.Category, "")
 	page := pointer.IntDeref(input.Page, 1)
 	pageSize := pointer.IntDeref(input.PageSize, 10)
-	res, err := c.Resource(common.SchemaOf(&common.ArcadiaAPIGroup, "Application")).Namespace("").List(ctx, metav1.ListOptions{
-		LabelSelector: v1alpha1.AppPublicLabelKey,
-	})
-	if err != nil {
+	res := &v1alpha1.ApplicationList{}
+	l := labels.Set{v1alpha1.AppPublicLabelKey: ""} // FIXME
```
Comment (Abirdcfly) [inline apiserver/pkg/gpt/gpt.go:101]: doubele check

File: apiserver/pkg/modelservice/modelservice.go [hunk 20]
```diff
@@ -40,7 +40,7 @@ import (
 // CreateModelService creates a 3rd_party model service
 // If serviceType is llm,embedding,then a LLM and a Embedder will be created
 // - Wrap all elements into *generated.ModelService
-func CreateModelService(ctx context.Context, c dynamic.Interface, input generated.CreateModelServiceInput) (*generated.ModelService, error) {
+func CreateModelService(ctx context.Context, c client.Client, input generated.CreateModelServiceInput) (*generated.ModelService, error) {
```
Comment (Abirdcfly) [inline apiserver/pkg/modelservice/modelservice.go:43]: change to  _

File: apiserver/pkg/knowledgebase/knowledgebase.go [hunk 21]
```diff
@@ -235,77 +218,75 @@ func UpdateKnowledgeBase(ctx context.Context, c dynamic.Interface, input *genera
...
 	if err != nil {
 		return nil, err
 	}
-	return knowledgebase2model(ctx, c, u)
+	return knowledgebase2model(ctx, c, kb)
 }
 
-func ListKnowledgeBases(ctx context.Context, c dynamic.Interface, input generated.ListKnowledgeBaseInput) (*generated.PaginatedResult, error) {
-	labelSelector, fieldSelector := "", ""
+func ListKnowledgeBases(ctx context.Context, c client.Client, input generated.ListKnowledgeBaseInput) (*generated.PaginatedResult, error) {
```
Comment (Abirdcfly) [inline apiserver/pkg/knowledgebase/knowledgebase.go:255]: todo change to common list input

File: apiserver/pkg/worker/worker.go [hunk 22]
```diff
@@ -144,7 +141,7 @@ func Worker2model(ctx context.Context, c dynamic.Interface, obj *unstructured.Un
 			w.ModelTypes = model.Types
 		}
 		w.Model = generated.TypedObjectReference{
-			APIGroup:  &common.ArcadiaAPIGroup,
+			APIGroup:  typedModel.APIGroup,
```
Comment (Abirdcfly) [inline apiserver/pkg/worker/worker.go:144]: doubel check

File: apiserver/pkg/worker/worker.go [hunk 23]
```diff
@@ -340,98 +328,72 @@ func UpdateWorker(ctx context.Context, c dynamic.Interface, input *generated.Upd
...
+		return nil, err
 	}
-
-	return nil, nil
+	err = c.DeleteAllOf(ctx, &v1alpha1.Worker{}, opts...)
+	return nil, err
 }
 
-func ListWorkers(ctx context.Context, c dynamic.Interface, input generated.ListWorkerInput, showWorkerModel bool, listOpts ...common.ListOptionsFunc) (*generated.PaginatedResult, error) {
+func ListWorkers(ctx context.Context, c client.Client, input generated.ListWorkerInput, showWorkerModel bool, listOpts ...common.ListOptionsFunc) (*generated.PaginatedResult, error) {
```
Comment (Abirdcfly) [inline apiserver/pkg/worker/worker.go:355]: todo change to list common input

File: apiserver/pkg/model/model.go [hunk 24]
```diff
@@ -81,172 +81,134 @@ func obj2model(obj *unstructured.Unstructured) (*generated.Model, error) {
...
+	if err != nil {
+		return nil, err
 	}
-	return nil, nil
+	err = c.DeleteAllOf(ctx, &v1alpha1.Model{}, opts...)
+	return nil, err
 }
 
-func ListModels(ctx context.Context, c dynamic.Interface, input generated.ListModelInput) (*generated.PaginatedResult, error) {
+func ListModels(ctx context.Context, c client.Client, input generated.ListModelInput) (*generated.PaginatedResult, error) {
```
Comment (Abirdcfly) [inline apiserver/pkg/model/model.go:133]: todo change to common list opts

File: apiserver/pkg/rag/rag.go [hunk 25]
```diff
@@ -369,24 +369,17 @@ func CreateRAG(ctx context.Context, kubeClient dynamic.Interface, input *generat
 		rag.Spec.Suspend = *input.Suspend
 	}
 
-	u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(rag)
-	if err != nil {
-		return nil, err
-	}
-	u1, err := kubeClient.Resource(schema.GroupVersionResource{Group: evav1alpha1.Group, Version: evav1alpha1.Version, Resource: "rags"}).Namespace(input.Namespace).Create(ctx, &unstructured.Unstructured{Object: u}, metav1.CreateOptions{})
+	err := kubeClient.Create(ctx, rag)
```
Comment (Abirdcfly) [inline apiserver/pkg/rag/rag.go:372]: double check create ns

File: apiserver/pkg/versioneddataset/versioned_dataset.go [hunk 26]
```diff
@@ -194,90 +186,83 @@ func VersionFiles(ctx context.Context, c dynamic.Interface, input *generated.Ver
...
-	}
+func ListVersionedDatasets(ctx context.Context, c client.Client, input *generated.ListVersionedDatasetInput) (*generated.PaginatedResult, error) {
 	page, size := 1, 10
 	if input.Page != nil && *input.Page > 0 {
 		page = *input.Page
 	}
 	if input.PageSize != nil && *input.PageSize > 0 {
 		size = *input.PageSize
 	}
-	ns := "default"
```
Comment (Abirdcfly) [inline apiserver/pkg/versioneddataset/versioned_dataset.go:216]: need this

File: apiserver/pkg/client/client.go (hunk 3 above)
Comment (Abirdcfly) [reply apiserver/pkg/client/client.go:51]: No need

File: apiserver/pkg/client/client.go (hunk 4 above)
Comment (Abirdcfly) [reply apiserver/pkg/client/client.go:74]: Done

File: pkg/evaluation/evaluation.go (hunk 7 above)
Comment (Abirdcfly) [reply pkg/evaluation/evaluation.go:74]: Done

File: pkg/evaluation/evaluation.go (hunk 8 above)
Comment (Abirdcfly) [reply pkg/evaluation/evaluation.go:153]: Done

File: apiserver/pkg/knowledgebase/knowledgebase.go [hunk 27]
```diff
@@ -60,7 +57,7 @@ func knowledgebase2model(ctx context.Context, c dynamic.Interface, obj *unstruct
 		status = "Deleting"
 	}
 
-	apiversion := obj.GetAPIVersion()
+	apiversion := knowledgebase.APIVersion
```
Comment (Abirdcfly) [inline apiserver/pkg/knowledgebase/knowledgebase.go:60]: doubel check

**Review by bjwswang — APPROVED** [review]

**Pull request merged**